### DISCUSSION TITLE
With ISR, should getStaticProps also cache in development mode?

### DISCUSSION THREAD

**Discussion by jylin** [vercel/next.js#32827] [Help]
With the introduction of ISR and the "revalidate" property, should getStaticProps also (maybe optionally) be able to cache the results of getStaticProps similar to how it behaves in production?

Currently, in development mode, getStaticProps seems to always re-render on every request. I think that makes sense, but now with the introduction of "revalidate", should we also have the option to have it cache in dev as well? I don't want to be making expensive calls even in dev, and to avoid that I would need to add code to also lookup some cache if NextJS is in dev mode. The behavior I want is exactly what "revalidate" provides (only make the expensive call once every X seconds and serve from the cache otherwise).

**Comment by balazsorban44** [comment]
Just a suggestion, but you could capture the result of such an expensive call in dev, and don't call the api at all. just use the data directly?

**Reply by jylin** [reply]
Possible, but the "revalidate" behavior is example what I want, so I'm hoping it's possible to enable that feature in dev too. If there's strong opposition, then maybe it should be at least documented that "revalidate" and getStaticProps will always re-render on every request in dev.
